src/utilities/safeexec.py

```diff
@@ -226,11 +226,17 @@ def add_dir(path, read_only, volumes):
         cmd += add_dir(d, True, volumes)
 
     if settings.DOCKER_CONTAINER_EXTERNAL_DIR is not None:
-        external_dir = settings.DOCKER_CONTAINER_EXTERNAL_DIR
-        task_id_custom = environment_variables.get('TASK_ID_CUSTOM')
-        if task_id_custom is not None and task_id_custom != "":
-            external_dir = string.Template(settings.DOCKER_CONTAINER_EXTERNAL_DIR).substitute(TASK_ID_CUSTOM=environment_variables.get('TASK_ID_CUSTOM'))
-        cmd += [f"--volume={external_dir}:/external:ro"]
+        external_dir = None
+        tmpl = string.Template(settings.DOCKER_CONTAINER_EXTERNAL_DIR)
+        requires_task_id_custom = "TASK_ID_CUSTOM" in tmpl.get_identifiers()
```

Comment:
small improvement: define a variable for the string "TASK_ID_CUSTOM" to avoid duplication.

Reply:
Fixed